File: Snooper/src/main/java/com/prateekj/snooper/networksnooper/database/HttpCallRecordCursorParser.java
package com.prateekj.snooper.networksnooper.database;

import android.database.Cursor;

import com.prateekj.snooper.database.CursorParser;
import com.prateekj.snooper.networksnooper.model.HttpCallRecord;

import java.util.Date;

import static android.provider.BaseColumns._ID;
import static com.prateekj.snooper.networksnooper.database.HttpCallRecordContract.COLUMN_DATE;
import static com.prateekj.snooper.networksnooper.database.HttpCallRecordContract.COLUMN_ERROR;
import static com.prateekj.snooper.networksnooper.database.HttpCallRecordContract.COLUMN_METHOD;
import static com.prateekj.snooper.networksnooper.database.HttpCallRecordContract.COLUMN_PAYLOAD;
import static com.prateekj.snooper.networksnooper.database.HttpCallRecordContract.COLUMN_RESPONSE_BODY;
import static com.prateekj.snooper.networksnooper.database.HttpCallRecordContract.COLUMN_STATUSCODE;
import static com.prateekj.snooper.networksnooper.database.HttpCallRecordContract.COLUMN_STATUSTEXT;
import static com.prateekj.snooper.networksnooper.database.HttpCallRecordContract.COLUMN_URL;

public class HttpCallRecordCursorParser implements CursorParser<HttpCallRecord> {

  @Override
  public HttpCallRecord parse(Cursor cursor) {
    HttpCallRecord httpCallRecord = new HttpCallRecord();
    httpCallRecord.setId(cursor.getLong(cursor.getColumnIndex(_ID)));
    httpCallRecord.setUrl(cursor.getString(cursor.getColumnIndex(COLUMN_URL)));
    httpCallRecord.setPayload(cursor.getString(cursor.getColumnIndex(COLUMN_PAYLOAD)));
    httpCallRecord.setResponseBody(cursor.getString(cursor.getColumnIndex(COLUMN_RESPONSE_BODY)));
    httpCallRecord.setMethod(cursor.getString(cursor.getColumnIndex(COLUMN_METHOD)));
    httpCallRecord.setStatusCode(cursor.getInt(cursor.getColumnIndex(COLUMN_STATUSCODE)));
    httpCallRecord.setStatusText(cursor.getString(cursor.getColumnIndex(COLUMN_STATUSTEXT)));
    httpCallRecord.setDate(new Date(cursor.getLong(cursor.getColumnIndex(COLUMN_DATE))));
    httpCallRecord.setError(cursor.getString(cursor.getColumnIndex(COLUMN_ERROR)));
    return httpCallRecord;
  }
}

File: Snooper/src/main/java/com/prateekj/snooper/networksnooper/database/HttpCallRecordCursorParser.kt
package com.prateekj.snooper.networksnooper.database

import android.database.Cursor

import com.prateekj.snooper.database.CursorParser
import com.prateekj.snooper.networksnooper.model.HttpCallRecord

import java.util.Date

import android.provider.BaseColumns._ID
import com.prateekj.snooper.networksnooper.database.HttpCallRecordContract.Companion.COLUMN_DATE
import com.prateekj.snooper.networksnooper.database.HttpCallRecordContract.Companion.COLUMN_ERROR
import com.prateekj.snooper.networksnooper.database.HttpCallRecordContract.Companion.COLUMN_METHOD
import com.prateekj.snooper.networksnooper.database.HttpCallRecordContract.Companion.COLUMN_PAYLOAD
import com.prateekj.snooper.networksnooper.database.HttpCallRecordContract.Companion.COLUMN_RESPONSE_BODY
import com.prateekj.snooper.networksnooper.database.HttpCallRecordContract.Companion.COLUMN_STATUSCODE
import com.prateekj.snooper.networksnooper.database.HttpCallRecordContract.Companion.COLUMN_STATUSTEXT
import com.prateekj.snooper.networksnooper.database.HttpCallRecordContract.Companion.COLUMN_URL

class HttpCallRecordCursorParser : CursorParser<HttpCallRecord> {

  override fun parse(cursor: Cursor): HttpCallRecord {
    return HttpCallRecord().apply {
      id = cursor.getLong(cursor.getColumnIndexOrThrow(_ID))
      url = cursor.getString(cursor.getColumnIndexOrThrow(COLUMN_URL))
      payload = cursor.getString(cursor.getColumnIndexOrThrow(COLUMN_PAYLOAD))
      responseBody = cursor.getString(cursor.getColumnIndexOrThrow(COLUMN_RESPONSE_BODY))
      method = cursor.getString(cursor.getColumnIndexOrThrow(COLUMN_METHOD))
      statusCode = cursor.getInt(cursor.getColumnIndexOrThrow(COLUMN_STATUSCODE))
      statusText = cursor.getString(cursor.getColumnIndexOrThrow(COLUMN_STATUSTEXT))
      date = Date(cursor.getLong(cursor.getColumnIndexOrThrow(COLUMN_DATE)))
      error = cursor.getString(cursor.getColumnIndexOrThrow(COLUMN_ERROR))
    }
  }
}
